salt/states/file.py

```diff
@@ -6754,7 +6754,9 @@ def copy_(name,
         elif not __opts__['test'] and changed:
             # Remove the destination to prevent problems later
             try:
-                __salt__['file.remove'](name)
+                # On windows, if a file has the read-only attribute then we are unable
+                # to complete this copy unless force is set to true.
+                __salt__['file.remove'](name, force=force)
```

Comment:
You'll probably have to gate this for Windows as the Linux version of `file.remove` does not include the `force` option.

```
if salt.utls.platform.is_windows():
  __salt__['file.remove'](name, force=force)
else:
  __salt__['file.remove'](name)
endif
```
Something like that ^^^

Reply:
Or we could modify the linux `file.remove` to accept `kwargs`

Reply:
This is probably what's causing all the test failures

Reply:
Ahh ok. Thx. I'll add this in a minute.

Reply:
Hey, actually this pattern of calling `file.remove` without the `force` parameter exists in a number of places within this state and so I imagine a number of them will fail with the read-only attribute set.

Is it worth fixing all of these now to solve this condition? Or is it better just to fix them as bugs get reported?

Reply:
Also I do believe that instead of adding the gate as you mentioned, that it'd be better to add the `force` parameter to `file.remove` as at some point posix acls might interfere with the api on linux (?) and so the semantics of `force` might come into existence on non-windows platforms.

It looks like some prior work also does the gating you suggested too.